linalg/blockoperator.cpp

```diff
@@ -29,12 +30,21 @@ BlockOperator::BlockOperator(const Array<int> & offsets)
      coef(nRowBlocks, nColBlocks)
 {
    op = static_cast<Operator *>(NULL);
-   row_offsets.MakeRef(offsets);
-   col_offsets.MakeRef(offsets);
+   if (owns_offsets)
+   {
+      row_offsets = offsets;
+      col_offsets = offsets;
+   }
+   else
+   {
+      row_offsets.MakeRef(offsets);
+      col_offsets.MakeRef(offsets);
+   }
```

Comment:
I agree it is a bit confusing. The current behavior is to reference the offset values. I was running into issues in this PR and other code managing the lifetime of `offsets` since this approach is prone to dangling references. This adds the option to simply copy the values so that `BlockOperator` owns the data.

Unless there's code that relies on the reference feature, maybe the behavior could be switched to always copy the offsets? I don't think much is gained using references since `offsets` will usually be small, and `Operator::height` is initialized from `offsets`, so changing `offsets` elsewhere could put the operator in an incorrect state.

Reply:
@YohannDudouit I can roll back this change and some others, like the ones in `operator`, that aren't exactly required for this PR, and submit them as separate smaller PRs.

Reply:
This is fairly minor to me, I wanted to bring attention onto it, but this is not critical to me, if you need this why not. I typically don't like to expose this kind of implementation detail, but if `MakeRef` is not doing the right thing for you, then I'm not sure there is an alternative.

Reply:
I agree we probably shouldn't make the interface more complicated by adding another parameter, so I reverted the change. It's not super related to this PR which is already kind of complicated. Seems appropriate for me to think about and put in a separate PR. If you disagree, I can put the change back in. As a side note I also reverted my changes to add transpose to`mfem::ConstrainedOperator` because I did not put any testing for it and it is no longer needed for my PR.

Reply:
Marking this as resolved because I reverted these changes.